Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -120,61 +146,161 @@ namespace gdjs {
+        }, 5000);
+      };
+
+      export const displayLeaderboard = function (
+        runtimeScene: gdjs.RuntimeScene,
+        leaderboardId: string,
+        displayLoader: boolean
+      ) {
+        _leaderboardViewIframeErrored = false;
+        _leaderboardViewIframeLoaded = false;
+        _leaderboardViewIframeLoading = true;
+        const gameId = gdjs.projectData.properties.projectUuid;
+        const targetUrl = `http://localhost:3001/games/${gameId}/leaderboard/${leaderboardId}?inGameEmbedded=true&dev=true`;
+        checkLeaderboardAvailability(targetUrl)
+          .then((isAvailable) => {
+            if (!isAvailable) {
+              onError(
+                runtimeScene,
+                'Leaderboard data could not be fetched. Doing nothing.'
+              );
+              return;
+            }
+
+            if (_leaderboardViewIframe) {
+              // Do not display loader when changing leaderboard
+              resetErrorTimeout(runtimeScene);
+              _leaderboardViewIframe.src = targetUrl;
+            } else {
+              const domElementContainer = runtimeScene
+                .getGame()
+                .getRenderer()
+                .getDomElementContainer();
+              if (!domElementContainer) {
+                onError(
+                  runtimeScene,
+                  "The div element covering the game couldn't be found, the leaderboard cannot be displayed."
+                );
+                return;
+              }
+
+              const iframe = document.createElement('iframe');
+
+              resetErrorTimeout(runtimeScene);
+              iframe.src = targetUrl;
+              iframe.id = 'leaderboard-view';
+              iframe.style.position = 'absolute';
+              if (displayLoader) {
+                // To trigger iframe loading and be able to listen to its events, use `opacity: 0` instead of `visibility: hidden` or `display: none`
+                iframe.style.opacity = '0';
+              }
+              iframe.style.pointerEvents = 'all';
+              iframe.style.top = '0px';
+              iframe.style.height = '100%';
+              iframe.style.left = '0px';
+              iframe.style.width = '100%';
+              iframe.style.border = 'none';
+              _leaderboardViewIframe = iframe;
+              if (typeof window !== 'undefined') {
+                _leaderboardViewClosingCallback = (event: MessageEvent) => {
+                  receiveMessage(runtimeScene, displayLoader, event);
+                };
+                (window as any).addEventListener(
+                  'message',
+                  _leaderboardViewClosingCallback,
+                  true
+                );
+              }
+              if (displayLoader) {
+                domElementContainer.appendChild(_loaderContainer);
+              }
+              domElementContainer.appendChild(_leaderboardViewIframe);
+            }
+          })
+          .catch(() => {
```

Comment:
These .catch after a then will catch errors launched by `checkLeaderboardAvailability`, but also by the entire body of the `then`. If it's not meant to do that, consider using then 2nd argument instead: `.then(() => {...}, (error) => {...})`.
(Catching everything seems like a good idea, but could also hide real errors like syntax errors, unhandled exception and generally it's better to let them just go rather than hiding them behind an error message that is unrelated and confusing)